USSF-ORBIT/ussf-portal
# Port e2e tests to separate repo

# SC-722

## Description 

Initial port of e2e tests to the portal repo. First step in moving the e2e tests out into a separate repo. To avoid cluttering the repo I created a folder called `e2e` to store all the e2e tests in.

## Review Notes

Also this should resolve SC-723 as it was easy enough to include in this PR.

Please review the github settings to ensure none were missed, though I copied it from our client repo.

## How to run the tests

Stop other running containers and remove all stopped containers

```sh
cd e2e
docker ps -qa | xargs docker rm -f
```

Start the system in docker. Note `--build` forces the building of the images, `--force-recreate` will remove the containers if they exist already, and `--remove-orphans` will remove any other running containers.

```sh
docker compose up --build --force-recreate --remove-orphans
```
Note: I've had a couple times where the CMS doesn't start up properly, I'm not clear why but ctrl-C and running the above command a second time works.

Once the above services are running you should run all the cypress & playwright tests. Everything should pass, though you might have to re-run things...

```sh
yarn
```

#### Cypress Based Tests

```sh
yarn cypress:install
yarn cypress:dev
```

#### Playwright Based Tests

These are still named `e2e` as I think the desire is to eventually migrate all our tests from cypress to playwright.

```sh
yarn e2e:install
yarn e2e:test
```
